Explorer/Assets/Scripts/SceneRuntime/V8EngineFactory.cs

```diff
@@ -1,10 +1,16 @@
-﻿using Microsoft.ClearScript.V8;
+﻿using DCL.DebugUtilities;
+using DCL.Diagnostics;
+using Microsoft.ClearScript.V8;
+using System.Collections.Concurrent;
 
 namespace SceneRuntime
 {
     public class V8EngineFactory
     {
-        public static V8ScriptEngine Create()
+        private readonly ConcurrentDictionary<SceneShortInfo, V8ScriptEngine> activeEngines = new ();
```

Comment:
`Factory` should remain a factory. Consider moving storage into a separate cache class

Reply:
extracted it in the inheritor of `ConcurrentDictionary`, called `V8ActiveEngines`